615 Update user select screen

MarcellinoCO opened the pull request:
# #615 Update User Select screen content to new design

This changes updated the UserSelectScreen to the new design provided. Here is the screenshot:
<img src="https://user-images.githubusercontent.com/19645892/117110171-ef439d80-adaf-11eb-88d8-805620a2e514.gif" height="500">

## Some Issues Faced:
1. The Box component for the profile picture is not matching its parent width in some devices (as seen in the screenshot), while in other devices it seems fine.
2. I can't seem to find a way to make the UserButton to wrap its content. I need to hardcode the size with the Modifier. I think it would be nicer to make the button to wrap its content.
3. Need to update 'Plant' icon under user's phone number (the icon can't be exported in the design).

## Where To Go From Here:
1. Need to fetch real user data (profile picture, name, phone, and number of tokens).
2. Update the bottom bar of the layout.

Elforama commented on app/src/main/java/org/greenstand/android/TreeTracker/userselect/UserSelectScreen.kt:
```diff
@@ -71,22 +81,91 @@ fun UserSelectScreen(
-            Text("User Select")
-            state.planters.forEach { user ->
+            // User profile picture.
+            Box(  // TODO: Fetch user profile picture.
```
To make this match parent you might be able to use fillMaxWidth and set its aspect ratio to 1:1

Elforama commented on app/src/main/java/org/greenstand/android/TreeTracker/userselect/UserSelectScreen.kt:
```diff
@@ -71,22 +81,91 @@ fun UserSelectScreen(
-                            viewModel.selectUser(user)
-                        }
-                        .background(color = if (state.selectedPlanter?.id == user.id) Color.Gray else Color.White)
+                    text = "Name\nPhone",  // TODO: Fetch user name and phone number.
```
We have access to the name already, let's use that here to keep the functionality.

Elforama commented on app/src/main/java/org/greenstand/android/TreeTracker/userselect/UserSelectScreen.kt:
```diff
@@ -71,22 +81,91 @@ fun UserSelectScreen(
+@Composable
+fun UserButton(
+    user: PlanterInfoEntity,
+    state: UserSelectState,
```
We're only passing in the screens state object to know if it's selected. To decouple the list item from the screen I'd suggest passing in an isSelected boolean instead.

Elforama reviewed (changes requested):
I left a few comments to address, but overall this looks great!

I'll write up some tasks for all the TODOs that you created.

MarcellinoCO commented:
This commit addresses the change requests from @Elforama. Here is the newest screenshot:
<img src="https://user-images.githubusercontent.com/19645892/117340674-573ad680-aecb-11eb-9781-1ceb9645bf89.gif" height="500">

I have solved the issues from my first comment, but have not done the Plant icon.
>     3. Need to update 'Plant' icon under user's phone number (the icon can't be exported in the design).

## Changes

**DepthButton changes**:
- Added `contentAlignment` parameter. This parameter is used to position the contents alignment inside the button.
- Changed the `DepthSurface` modifier's `fillMaxSize()` into `matchParentSize()`. This allows the `DepthSurface` to match its size to the contents of the button, and the use of `wrapContentSize()` in the modifier parameter.
- Added documentation to the component.

**UserButton changes**:
- Removed explicit button height, and used `wrapContentHeight()` instead.
- Decoupled the `UserSelectState`, replacing it with a boolean instead.

Elforama commented:
Great fixes, this is looking good.
